### ISSUE TITLE
Cilium agent can crash during startup if ENI is not attached yet

### ISSUE THREAD

**Issue by gandro** [cilium/cilium#16494]
In ENI IPAM mode, there is a race that can happen where the operator to attaches the ENI to the instance and updates the CiliumNode CRD, but when the agent tries to find the ENI to allocate the router IP, the Linux network interface has not been attached yet. This causes Cilium agent to fail with the following error message:

```
level=fatal msg="Error while creating daemon" error="Unable to allocate router IP for family ipv4: unable to find ENI eni-0c0a52045c831422b" subsys=daemon
```

Cilium will recover after a few attempts (usually, it doesn't take more than 5 seconds for the ENI to be available on the node), but this fatal error leads to false positives in monitoring and therefore should be fixed. 

For ENIs attached later in the life-cycle of a Cilium pod, we wait for the ENI devices to be attached before we try to configure them:
https://github.com/cilium/cilium/blob/4f27865e975cc095891b68df5f4a7bc0e26ecca5/pkg/ipam/crd_eni.go#L76-L77

**Comment by gandro** [comment]
Upon closer inspection, this particular error here is not caused by the ENI device not being attached. The error I had in mind that would be fixed by some form of `waitForNetlinkDevices` would be this one here:
https://github.com/cilium/cilium/blob/5839d2322f3b691e419fcad25a01c29373d96996/pkg/datapath/loader/base.go#L171

What's happening on the above log line however is different: It happens when the IPAM allocation result does not match the CiliumNode spec. This is unexpected, as we don't remove ENI devices from VMs, so the CiliumNode should always contain all ENIs from which IPs are allocated.

Given that in the above example, the ENI seemed indeed attached to the VM according to the AWS console. Not clear why it was temporarily removed from the CiliumNode.

**Comment by codablock** [comment]
fyi, #18352 is slightly related as the fixed bug results from the same underlying issue observed here

**Comment by joestringer** [comment]
@pchaigno I'm guessing based on the lack of activity since May that you aren't actively planning to investigate this? I've dropped the backport labels since it doesn't look like this will make it before those versions go out of support.

**Comment by pchaigno** [comment]
@gandro Wasn't this fixed by you in https://github.com/cilium/cilium/pull/19923?

**Comment by gandro** [comment]
> @gandro Wasn't this fixed by you in #19923?

Oh, true! Totally forgot about it. But yes, I believe that PR fixed this issue. Closing!
